doc/developer-documentation.md

```diff
@@ -17,105 +17,147 @@ In order to enable a new tree, we need to add an entry there.
 ```
 
-After adding a `trees` entry, we need to define build configurations for it.
-In the same [configuration](https://github.com/kernelci/kernelci-pipeline/blob/main/config/pipeline.yaml) file, `build_configs` section is there to specify them.
-For example, we need to specify which branch to monitor of a particular tree and other build variants as well.
+This `tree-name` is used to refer to this tree from here onwards.
+After adding a `trees` entry, we need to define build and test configurations for it. In the same [configuration](https://github.com/kernelci/kernelci-pipeline/blob/main/config/pipeline.yaml) file, `jobs` section is there to specify them. `ChromeOS` specific job definitions are located in [config/jobs-chromeos.yaml](https://github.com/kernelci/kernelci-pipeline/blob/main/config/jobs-chromeos.yaml) file. Depending upon the type of the job such as kbuild or test job, different parameters are specified:
 
 For instance,
-```yaml
-build_configs:
-
-    kernelci_staging-mainline:
-        tree: kernelci
-        branch: 'staging-mainline'
-        variants:
-            gcc-10:
-                build_environment: gcc-10
-                architectures:
-                    x86_64:
-                        base_defconfig: 'x86_64_defconfig'
-                        filters:
-                            - regex: { defconfig: 'x86_64_defconfig' }
-```
-
-That's it! The tree is enabled now.
-All the jobs defined under `jobs` section of [config file](https://github.com/kernelci/kernelci-pipeline/blob/main/config/pipeline.yaml) would run on the newly added tree until specified otherwise.
-
-
-## Enabling a new test
-
-KernelCI currently runs several test suites. This manual is intended
-to provide documentation for developers on how to enable new tests.
-
-
-### Job definition
-The pipeline [configuration](https://github.com/kernelci/kernelci-pipeline/blob/main/config/pipeline.yaml) file has `jobs` section for defining various build and test jobs. The currently enabled tests are `kver`, `baseline`, `kunit`, `kselftest`, and `sleep` tests.
-`ChromeOS` specific job definitions are located in [config/jobs-chromeos.yaml](https://github.com/kernelci/kernelci-pipeline/blob/main/config/jobs-chromeos.yaml) file.
-
-To enable a new test, an entry needs to be added as follows:
-
 ```yaml
 jobs:
-  <job-name>:
+
+  <kbuild-job-name>:
     template: <job-template>
-    kind: <kind of job, either 'kbuild', 'job' or 'test'>
+    kind: kbuild
     image: <docker-image-name>
-    params: <parameters such as arch, compiler, defconfig>
-    rules: <job-rules>
+    params:
+      arch: <architecture>
+      compiler: <compiler_name>
+      cross_compile: <compiler_prefix>
+      dtbs_check: <dtb_check_enabled>
+      defconfig: <defconfig_name>
+      fragments:
+       - <fragment_name>
+    rules:
+      min_version:
+        version: <kernel_version>
+        patchlevel: <kernel_patch_level>
+      tree:
+        - <tree_name>
+        - <!tree_name>
```

Comment:
I think it should be the below instead
```suggestion
        - !<tree_name>
```
Also, I don't think the same `<tree-name>` should be there for both the rules.

Reply:
The above suggestion has not been addressed yet to exclude `!` from `<tree-name>`.

Reply:
Also, it should be explained somewhere what `!` mark means in `rules`.

Reply:
I'd updated to 
```
      tree:
        - <tree_name>
        - <!tree_name2>
```

I want to give an example of how we can include a tree `<tree_name>` and ignore a tree `<!tree_name2>`. Technically is it correct that we can specify two trees like this? What do you suggest?

Reply:
Yes, you can definitely include two tree names with different rules. I am suggesting to keep `!` outside of `<tree-name2>` marker.
Please take a look at https://github.com/kernelci/kernelci-pipeline/pull/686#discussion_r1682244359 for suggested change.

Reply:
Now I've realized that ! should be outside <>. Thanks for mentioning it again. Fixing.